src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -136,7 +166,7 @@ ObjectDataHandler::write_ret do_insertions(
     to_write,
     [ctx](auto &region) {
       LOG_PREFIX(object_data_handler.cc::do_insertions);
-      if (region.to_write) {
+      if (region.type == extent_to_write_t::type_t::DATA) {
```

Comment:
Might be worth replacing these comparisons with is_data(), is_zero(), is_existing()?

Reply:
done